python_modules/automation/generate_synthetic_events.py

```diff
@@ -0,0 +1,160 @@
+            'user_create',
+            'user_delete',
+            'signup',
+        ]
+        return 'io.dagster.{}'.format(random.choice(event_types))
+
+    def _gen_timestamp(self):
+        midnight = datetime.datetime.combine(
+            self.start_date, datetime.time.min, tzinfo=datetime.timezone.utc
+        ).timestamp()
+        return midnight + random.randint(0, 86400 - 1)
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        return json.dumps(
+            {
+                'environment': 'production',
+                'method': 'GET',
+                # Nested dicts
+                'cookies': {
+                    'session': secrets.token_urlsafe(16),
+                    'persistent': secrets.token_urlsafe(16),
+                },
+                'run_id': self.faker.uuid4(),
+                'type': self._gen_event_type(),
+                'user_agent': self._gen_user_agent(),
+                'ip_address': self.faker.ipv4_public(),
+                'timestamp': self._gen_timestamp(),
+                'url': '/' + self.faker.uri_path(),
+                # like any good production system, we throw some random PII in our behavioral events
+                'name': self.faker.name(),
+                'email': self.faker.ascii_email(),
+                # Nested lists
+                'location': list(self.faker.location_on_land(coords_only=False)),
+            }
+        )
+
+
+def create_events_file_for_date(date, num_events_per_file):
+    eg = EventGenerator(date)
+
+    f = tempfile.NamedTemporaryFile(delete=False)
+    for _ in range(num_events_per_file):
+        f.write(str.encode(next(eg) + '\n'))
+    f.close()
+    return f.name
+
+
+def write_to_s3(filename_and_date, s3_bucket, s3_prefix, num_files):
+    filename, date = filename_and_date
+    s3_client = boto3.client('s3')
+    for _ in range(num_files):
+        s3_path = os.path.join(
+            s3_prefix, datetime.datetime.strftime(date, "%Y/%m/%d"), 'json-' + uuid.uuid4().hex
+        )
+        print("Writing events file to S3 path: %s" % s3_path)
+        s3_client.put_object(Body=open(filename, 'rb'), Bucket=s3_bucket, Key=s3_path)
+
+
+@click.command()
+@click.option('--s3-bucket', help='The S3 bucket to use', required=True)
+@click.option('--s3-prefix', help='The S3 prefix to use', required=True)
+@click.option(
+    '--start-date',
+    help='The starting UTC date (in format 2019-01-01) for which to generate events.',
+    required=True,
+)
+@click.option(
+    '--end-date',
+    help='The ending UTC date, exclusive (in format 2019-01-01) for which to generate events.',
+    required=True,
+)
+@click.option('--num-files', default=100, help='Total number of event files.', required=True)
+@click.option(
+    '--num-events-per-file',
+    default=100000,
+    help='Batch size, number of events per file.',
+    required=True,
+)
+@click.option('--dry-run', is_flag=True, help='Dry runs won\'t write to S3.')
```

Comment:
dry runs create the files and then delete them? why bother?

Reply:
ah good catch, I meant to put the os.unlink() under the `if not dry_run`, will update - meant to permit inspecting the files